python/torch_mlir_e2e_test/test_suite/reshape_like.py

```diff
@@ -657,6 +657,28 @@ def ViewNoChangeStaticModule_basic(module, tu: TestUtils):
 
 # ==============================================================================
 
+
+class AsStridedStaticModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([3, 3], torch.float32, True),
+    ])
+
+    def forward(self, x):
+        return torch.ops.aten.as_strided(x, (2, 2), (1, 2), 1)
```

Comment:
Does this implementation work if you do

```python
        return torch.ops.aten.as_strided(x.t(), (2, 2), (1, 2), 1)
```

I.e. pass the transpose of `x` as the argument

Reply:
To add to this, I think adding support for this op in torch-mlir will be very tricky. The reason is that this op depends on knowledge about the storage used by the input tensor, and at the torch dialect level in torch-mlir there is no notion of storage. In the example I gave above, `x.t()` does not change the storage of the tensor, so PyTorch returns the same output as when `x` is passed. However, I expect torch-mlir will output a tensor as if `x.t().contiguous()` had been passed instead because it does not know that `x.t()` and `x` share the same storage. 

Where is it that you're seeing this op used? Given the warning in the [documentation](https://pytorch.org/docs/stable/generated/torch.as_strided.html?highlight=as_strided#torch.as_strided) of the op, I would expect that this op is not explicitly used in the definition of a model, but rather it is being generated by PyTorch when turning the model to JIT IR. If this is the case, then maybe we can find a way to fold it back.

Reply:
With` x.t()`, it fail.
```
➜  torch-mlir git:(as_stride) ✗ python -m e2e_testing.main -c tosa -f "AsStridedStaticModule_basic"
FAIL - "AsStridedStaticModule_basic"

Unexpected outcome summary:

****** Failed tests - 1 tests
    FAIL - "AsStridedStaticModule_basic"

Summary:
    Failed: 1
```

Reply:
`it is being generated by PyTorch when turning the model to JIT IR.`
Not sure how to find if it is generated by JIT IR. And furthermore, if it is, how to fix it?